client/src/main/java/org/asynchttpclient/netty/request/NettyRequestSender.java

```diff
@@ -265,16 +265,10 @@ private Channel getOpenChannel(NettyResponseFuture<?> future, Request request, P
 
         Object partitionKey = future.getPartitionKey();
 
-        // we disable channelPreemption when performing next requests
-        final boolean acquireChannelLock = !performingNextRequest;
```

Comment:
The original author's intent was to not require an extra semaphore when being redirected to another host. IMHO, too specific to be worth supporting.